codemirror/codemirror-runmode.d.ts

```diff
@@ -0,0 +1,22 @@
+﻿// Type definitions for CodeMirror
+// Project: https://github.com/marijnh/CodeMirror
+// Definitions by: Joseph Vaughan <https://github.com/Joev->
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// See docs https://codemirror.net/doc/manual.html#addon_runmode
+
+declare module CodeMirror {
+
+    /**
+     * Can be used to run a CodeMirror mode over text without actually opening an editor instance.
+     *
+     * @param text The document to run through the highlighter.
+     * @param mode The mode to use (must be loaded as normal).
+     * @param output If this is a function, it will be called for each token with
+     *               two arguments, the token's text and the token's style class
+     *               (may be null for unstyled tokens). If it is a DOM node, the
+     *               tokens will be converted to span elements as in an editor,
+     *               and inserted into the node (through innerHTML).
+     */
+    function runMode(text : string, mode : any, output : any): void;
```

Comment:
Maybe let's use the argument names [from the original?](https://github.com/codemirror/CodeMirror/blob/master/addon/runmode/runmode.js#L14)

```javascript
CodeMirror.runMode = function(string, modespec, callback, options) {
```

Can also peg `callback` argument to a union type: `callback: (HTMLElement | (text: string, style: string) => void)`

Optional last argument can also be added something like: `options?: { tabSize?: number; state?: any; }` — although that one is probably rarely used so can skip for now.